sql/core/src/main/scala/org/apache/spark/sql/execution/python/PythonArrowInput.scala

```diff
@@ -194,3 +180,115 @@ private[python] trait BatchedPythonArrowInput extends BasicPythonArrowInput {
     }
   }
 }
+
+object BatchedPythonArrowInput {
+  /**
+   * Split a group into smaller Arrow batches within
+   * a separate and complete Arrow streaming format in order
+   * to work around Arrow 2G limit, see ARROW-4890.
+   *
+   * The return value is the number of rows in the batch.
+   * Each split Arrow batch also does not have mixed grouped. For example:
+   *
+   *        +------------------------+      +------------------------+      +--------------------
+   *        |Group (by k1) v1, v2, v3|      |Group (by k2) v1, v2, v3|      |                 ...
+   *        +------------------------+      +------------------------+      +--------------------
+   *
+   * +------+-----------------+------+------+-----------------+------+------+--------------------
+   * |Schema|            Batch| Batch|Schema|            Batch| Batch|Schema|           Batch ...
+   * +------+-----------------+------+------+-----------------+------+------+--------------------
+   * |    Arrow Streaming Format     |    Arrow Streaming Format     |    Arrow Streaming Form...
+   *
+   * Here, each (Arrow) batch does not span multiple groups.
+   * These (Arrow) batches within each complete Arrow IPC Format are
+   * reconstructed into the group back as pandas instances later on the Python worker side.
+   */
+  def writeSizedBatch(
```

Comment:
This is basically what `BaseStreamingArrowWriter` is already doing for `applyInPandasWithState` and `transformWithState`, except that never got updated to handle `maxBytesPerBatch`. Should this just be consolidated there so there's not yet another place that needs to know how to chunk up arrow streams?

Reply:
Make sense, created https://issues.apache.org/jira/browse/SPARK-53608 to track all related works

Reply:
For the reuse of `BaseStreamingArrowWriter`, since it is only used in pyspark streaming feature, I am still hesitant about using it here.
created subtask https://issues.apache.org/jira/browse/SPARK-53612 to track the consolidating it